packages/server/src/interfaces.ts

```diff
@@ -28,14 +28,31 @@ export interface Service {
   app: HttpApp
 }
 
-export interface Simulation {
-  id: string;
-  simulators: Record<string, Simulator>;
-  services: Service[];
-  addSimulator(name: string, simulator: Simulator): Simulation;
-  scope: Task;
+export interface ServerState {
+  simulations: Record<string, SimulationState>;
 }
 
+export type SimulationState =
```

Comment:
I love the smell of algebraic data types/discriminated untions in the morning :).  I was thinking along these lines also.